File: Address.Core/Address.cs
namespace Address.Core;

public class Address
{
    public FederalSubjectType SubjectType { get; set; }
    public string? District { get; set; }
    public string? Settlement { get; set; }
    public string? Locality { get; set; }
    public string? CityDistrict { get; set; }
    public string? PlanningElement { get; set; }
    public StreetType? Street { get; set; }
    public string? LandPlot { get; set; }
    public BuildingType? Building { get; set; }
    public string? Room { get; set; }
    public Address() {}
}

File: AddressKotlin/entities/Address.kt
package entities

import enums.PlanningStructureType
import enums.LandPlotType
import enums.StreetType
import enums.CityDistrictType
import enums.SettlementType
import enums.FederalSubjectType
import entities.Subject
import entities.Settlement
import entities.Locality
import entities.CityDistrict
import entities.Street
import entities.Building
import entities.LandPlot
import entities.Room

class Address(
    var countryName: String,
    var subject: Subject,
    var district: String,
    var settlement: Settlement,
    var locality: Locality,
    var cityDistrict: CityDistrict,
    var planningStructure: PlanningStructureType,
    var street: Street,
    var building: Building,
    var room: Room,
    var parkingSpaceNumber: String
)
